### PR TITLE
[BUGFIX] Patch issue with data docs page retrieval in checkpoint actions

### LINKED ISSUE #10363: [BUG] GX unable to send notification to Teams channel after data validation in the checkpoint
**Describe the bug**
Using Great expectations version 1.0.2, I created an action inside a checkpoint to send a notification to a teams channel through a webhook but It is unable to do that and displays this error:

**To Reproduce**
```python
**create data source, asset, batch definition and expectation suite**
dataframe = spark.sql("SELECT * FROM xxx")
context = gx.get_context(project_root_dir="/dbfs/xxx")

data_source_name = "my_data_source"
data_asset_name = "my_dataframe_data_asset"
batch_definition_name = "my_batch_definition"
data_source = context.data_sources.add_or_update_spark(name=data_source_name)
data_asset=data_source.add_dataframe_asset(name=data_asset_name)
batch_definition = data_asset.add_batch_definition_whole_dataframe(batch_definition_name)


suite_name = "my_expectation_suite"
suite = gx.ExpectationSuite(name=suite_name)
suite = context.suites.add(suite)

table_row_count=dataframe.count()
table_row_count_expectationn = gxe.ExpectTableRowCountToEqual(value=table_row_count)
suite.add_expectation(table_row_count_expectationn)

**Retrieve the expectation suite and create validation definition**
expectation_suite_name = "my_expectation_suite"
expectation_suite = context.suites.get(name=expectation_suite_name)

definition_name = "my_validation_definition"
validation_definition = gx.ValidationDefinition(
        data=batch_definition, suite=expectation_suite, name=definition_name)
validation_definition = context.validation_definitions.add(validation_definition)

 validation_definitions = [
    context.validation_definitions.get("my_validation_definition")]

validation_notification_microsoft_teams_webhook = "xxx"

**create an action list**
action_list = [
    # This Action sends a Team Notification if an Expectation fails.
    gx.checkpoint.MicrosoftTeamsNotificationAction(
        name="send_microsoft_teams_notification_on_validation_result",
        teams_webhook= validation_notification_microsoft_teams_webhook,
        notify_on= "all",
    ),
    # This Action updates the Data Docs static website with the Validation
    #   Results after the Checkpoint is run.
    gx.checkpoint.UpdateDataDocsAction(
        name="update_all_data_docs",
    ),
]

**create a checkpoint**
checkpoint_name = "my_checkpoint"
    checkpoint = gx.Checkpoint(
        name=checkpoint_name,
        validation_definitions=validation_definitions,
        actions=action_list,
        result_format={"result_format": "COMPLETE"},
    )
    context.checkpoints.add(checkpoint)


batch_parameters = {"dataframe": dataframe}

validation_results = checkpoint.run(
    batch_parameters=batch_parameters,)


**Below is the error message I get**
File /local_disk0/.ephemeral_nfs/envs/pythonEnv-7048a166-73a8-4fdf-aa6c-fb34a0b4805e/lib/python3.10/site-packages/great_expectations/render/renderer/microsoft_teams_renderer.py:114, in MicrosoftTeamsRenderer._render_data_docs_links(self, data_docs_pages)
    112 if docs_link_key == "class":
    113     continue
--> 114 docs_link = data_docs_pages[docs_link_key]
    115 report_element = self._get_report_element(docs_link)
    116 elements.append(report_element)

TypeError: list indices must be integers or slices, not ValidationResultIdentifier
```

**Expected behavior**
The checkpoint is expected to send a notification to a teams channel through the teams webhook provided.

**Environment:**
 - Operating System: Windows
 - Great Expectations Version: 1.0.2
 - Data Source: Spark Dataframe
 - Cloud environment: Databricks

### PR DESCRIPTION
**Pull request by cdkini**
```
File /local_disk0/.ephemeral_nfs/envs/pythonEnv-7048a166-73a8-4fdf-aa6c-fb34a0b4805e/lib/python3.10/site-packages/great_expectations/render/renderer/microsoft_teams_renderer.py:114, in MicrosoftTeamsRenderer._render_data_docs_links(self, data_docs_pages)
    112 if docs_link_key == "class":
    113     continue
--> 114 docs_link = data_docs_pages[docs_link_key]
    115 report_element = self._get_report_element(docs_link)
    116 elements.append(report_element)

TypeError: list indices must be integers or slices, not ValidationResultIdentifier
```

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/render/renderer/microsoft_teams_renderer.py [hunk 1]
```diff
@@ -23,9 +23,7 @@ class MicrosoftTeamsRenderer(Renderer):
     MICROSOFT_TEAMS_SCHEMA_URL = "http://adaptivecards.io/schemas/adaptive-card.json"
 
     @override
-    def render(
-        self, checkpoint_result: CheckpointResult, data_docs_pages: list[dict] | None = None
-    ):
+    def render(self, checkpoint_result: CheckpointResult, data_docs_pages: dict | None = None):
```
Comment (billdirks) [inline great_expectations/render/renderer/microsoft_teams_renderer.py]: Can you annotate the return type here?

File: tests/checkpoint/test_checkpoint.py [hunk 2]
```diff
@@ -921,6 +929,25 @@ def test_checkpoint_run_adds_requisite_ids(self, tmp_path: pathlib.Path):
         assert meta["checkpoint_id"] == checkpoint.id
         assert meta["validation_id"] == checkpoint.validation_definitions[0].id
 
+    @pytest.mark.filesystem
+    def test_checkpoint_run_with_data_docs_and_slack_actions_emit_page_links(
+        self, tmp_path: pathlib.Path
+    ):
+        actions = [
+            UpdateDataDocsAction(name="docs_action"),
+            SlackNotificationAction(name="slack_action", slack_webhook="webhook"),
```
Comment (billdirks) [inline tests/checkpoint/test_checkpoint.py:937]: Is there a test with a `SlackNotificationAction` and no `UpdateDataDocsAction`? That is, do we test that we can run a checkpoint with and without a data docs action when combined with another action?

Also, would it be useful to do this test with the actions sorted in the opposite order. I know we always run the data docs action first and we have some testing. I don't know if we have an e2e test for that.

**Review by billdirks — COMMENTED** [review]
Thanks for doing this. Had a question around testing.

File: tests/checkpoint/test_checkpoint.py (hunk 2 above)
Comment (cdkini) [reply tests/checkpoint/test_checkpoint.py:937]: Flipped the order and added a new test with just slack

**Review by billdirks — APPROVED** [review]
LGTM, thanks!

**Pull request merged**